### PR TITLE
test(e2e): migrate `oras manifest` specs to ZOT

### PR DESCRIPTION
**Pull request by qweeah**
Depends on merging of https://github.com/oras-project/oras/pull/1088.

### CODE REVIEW

File: test/e2e/suite/command/manifest.go [hunk 1]
```diff
@@ -172,108 +172,103 @@ var _ = Describe("1.1 registry users:", func() {
...
 		It("should fetch image descriptor with media type assertion and platform validation", func() {
-			ORAS("manifest", "fetch", RegistryRef(Host, ImageRepo, multi_arch.LinuxAMD64.Digest.String()), "--platform", "linux/amd64", "--media-type", "application/vnd.oci.image.manifest.v1+json", "--descriptor").
+			ORAS("manifest", "fetch", RegistryRef(ZOTHost, ImageRepo, multi_arch.LinuxAMD64.Digest.String()), "--platform", "linux/amd64", "--media-type", "application/vnd.oci.image.manifest.v1+json", "--descriptor").
 				MatchContent(multi_arch.LinuxAMD64DescStr).Exec()
 		})
 
-		It("should fail to fetch image if media type assertion fails", func() {
-			ORAS("manifest", "fetch", RegistryRef(Host, ImageRepo, multi_arch.LinuxAMD64.Digest.String()), "--media-type", "this.will.not.be.found").
-				ExpectFailure().
-				MatchErrKeyWords(multi_arch.LinuxAMD64.Digest.String(), "error: ", "not found").Exec()
```
Comment (Wwwsylvia) [inline test/e2e/suite/command/manifest.go:273]: Why is this not included in v1.1 tests?

Comment (qweeah) [reply test/e2e/suite/command/manifest.go:273]: because ZOT doesn't accept `Accept` header and there is no way to test in ZOT.

Comment (Wwwsylvia) [reply test/e2e/suite/command/manifest.go:273]: oh interesting...

**Review by Wwwsylvia — APPROVED** [review]
LGTM, but I am not a maintainer

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**